modules/apim-adapter/src/main/java/com/axway/apim/api/specification/OAS3xSpecification.java

```diff
@@ -127,11 +133,15 @@ public void updateServerSection(String backendBasePath, String serverUrl) throws
         ((ObjectNode) openAPI).set(SERVERS, mapper.createArrayNode().add(newServer));
     }
 
-    public void overrideServerSection(String backendBasePath) {
-        if (backendBasePath.endsWith("/"))
-            backendBasePath = backendBasePath.substring(0, backendBasePath.length() - 1);
-        LOG.info("overriding openapi Servers url with value : {}", backendBasePath);
-        ObjectNode newServer = createObjectNode("url", backendBasePath);
+    public void overrideServerSection(String backendBasePath, String serverUrl) throws MalformedURLException, URISyntaxException {
+        URL backendBasePathURL = new URL(backendBasePath);
+        URI serverURL = URI.create(serverUrl);
```

Comment:
the name comes from the backendBasepath field in API json config which is passed as a first parameter to method configureBasePath, so I kept the same name, do you have in mind a better name?

Reply:
I mean we dont need to make any changes here.  the existing code will take care of replacing the servers.url.  

I do see that you are appending basepath to the existing host name.  Actually we don't need this as we are replacing the whole url with backendbashpath (it is a confusing name; the value contains fully qualified url with host and port).

Reply:
mmm I think I missed the point, backendBasePath in json configuration file can contain only host and port but also host, port and relative url. If it contains only host and port the behaviour is that the relative url is taken directly from the swagger base path otherwise it overwrites what is defined in swagger (and if overrideSpecBasePath=true). 

Consider the following table screenshot with the test cases evaluating:
* value of backend base path in json config file
* values in swagger/oas
* value of property overrideSpecBasePath
and comparing the resulting Backend API URL on API Manager.

![image](https://github.com/Axway-API-Management-Plus/apim-cli/assets/15086835/f7655696-e124-41f8-a8b7-2991e10b631b)

OAS3 and swagger 2.X look coherent to each other. The changes in code go in this direction.

Does this table looks ok for you?
Is there a case that doesn't look good to you?
@rathnapandi